packages/components/src/hooks/index.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useSelector, useDispatch} from 'react-redux';
+
+import {GlobalState} from '@mattermost/types/store';
+import {ClientConfig, ClientLicense} from '@mattermost/types/config';
+import {PreferenceType} from '@mattermost/types/preferences';
+import {UserProfile} from '@mattermost/types/users';
+
+import {getUser as fetchUser} from 'mattermost-redux/actions/entities/users';
+
+import {useEffect} from 'react';
+
+import {getPreferenceKey} from '../utils/preferences';
+import Preferences from '../constants/preferences';
+
+function getConfig(state: GlobalState): Partial<ClientConfig> {
+    return state.entities.general.config;
+}
```

Comment:
Re-implemented some mattermost-redux selectors here that were needed below. This isn't really ideal since now this code is in two places. I see two options:

1. Import it from mattermost-redux (bad because we want to reduce mattermost-redux usage)
2. Move selectors like this into `components` and export them to mattermost-redux which can re-export them to whatever is still using them. Then, eventually when mattermost-redux dies, everything needing these selectors can import them from `components` (not great, but better?)

Can anyone think of a different option?

Reply:
Does the component need to have these, or could we pass in the values from outside? It's not ideal since the users of this library will still have to get the user's profile and `teammateNameDisplay` preference themselves, but I'd prefer holding off on accessing the redux store from within this library as long as we can. I know that's likely a "when" rather than an "if", but I'd be worried that once we start doing it this way, it'll be hard to switch to something cleaner once that exists, so I'd like to hold off on that for as long as possible.

Something that also might work is having the web app provide these sort of hooks through Context. That way, components will know how to get a user, but they won't know where it comes from.

Reply:
I do agree, I'd rather not have to bring actions and selectors in but I think the only way to pass them in through the outside would be to:

1. Not have any component hierarchy in the components library, and instead have "pure" (probably not the right word) components and then when they're used in a product, wrap them all with redux connector components and introduce the hierarchy there
2. Keep the hierarchy in the components library, but pass the props all the way down from the parent components at the top of the tree

I think I'd lean towards 1) otherwise we'll be doing a lot of props piping. Unless there's another option you can think of